packages/frames.js/src/next/server.tsx

```diff
@@ -351,14 +354,24 @@ export function FrameContainer<T extends FrameState = FrameState>({
 
             // set redirect data for retrieval
             if ((child.props as any).action === "post_redirect") {
-              redirectMap[nextIndexByComponentType.button] = (
-                child.props as any as FrameButtonPostRedirectProvidedProps
-              ).target!;
+              const target =
+                (child.props as any as FrameButtonPostRedirectProvidedProps)
+                  .target || null;
+              const buttonIndex = target
+                ? nextIndexByComponentType.button
+                : (`_${nextIndexByComponentType.button}` as `_${number}`);
+              redirectMap[buttonIndex] = target;
             }
 
             return (
               <FFrameButtonShim
                 {...(child.props as any)}
+                // Don't include target if action is post_redirect, otherwise the next message will be posted to the target url
```

Comment:
This is a fix for a pre-existing bug in the framework which would POST to the desired redirect url instead of redirecting to it when `action=post_redirect`.

For example, 
```tsx
<FrameButton action="post_redirect" target="https://google.com">Go to google.com</FrameButton>
```

The above code would make a POST request to "https://google.com" expecting it to return a url